lib/stripe/stripe_object.rb

```diff
@@ -184,5 +184,72 @@ def method_missing(name, *args)
     def respond_to_missing?(symbol, include_private = false)
       @values && @values.has_key?(symbol) || super
     end
+
+    protected
+
+    def serialize_nested(key)
+      new_value = @values[key]
+      if @unsaved_values.include?(key)
+        # the object has been reassigned
+        # i.e. as object.key = {foo => bar}
+        update = new_value
+        new_keys = update.keys.map(&:to_sym)
+        # remove keys at the server, but not known locally
+        keys_to_unset = @original_values[key].keys - new_keys
+        keys_to_unset.each {|key| update[key] = ''}
+
+        update
+      else
+        if new_value.is_a?(Array) && @original_values[key].length > @values[key].length
+          raise ArgumentError.new(
+            "You cannot delete an item from an array, you must instead set a new array"
+          )
+        end
+
+        # can be serialized normally
+        serialize_params(new_value)
+      end
+    end
+
+    def serialize_params(obj)
+      case obj
+      when nil
+        ''
+      when StripeObject
+        unsaved_keys = obj.instance_variable_get(:@unsaved_values)
+        obj_values = obj.instance_variable_get(:@values)
+        update_hash = {}
+
+        unsaved_keys.each do |k|
+          update_hash[k] = serialize_params(obj_values[k])
+        end
+
+        obj_values.each do |k, v|
+          if v.is_a?(StripeObject) || v.is_a?(Hash)
```

Comment:
The way `Util.flatten_params` works is that `foo: {}` turns into `''`, since there is no inner information to update.